lib/reservation.rb

```diff
@@ -0,0 +1,24 @@
+module Hotel
+  class Reservation
+    attr_reader :id, :start_date, :end_date, :room, :total_cost, :block_id, :discount
+
+    def initialize(id:, start_date:, end_date:, room:, total_cost: nil, block_id: nil, discount: nil)
+      @id = id
+      @start_date = start_date
+      @end_date = end_date
+      raise ArgumentError if end_date <= start_date
+      @room = room
+      @block_id = block_id
+      @discount = discount
+      @total_cost = calc_total_cost
+    end
```

Comment:
In stead of storing `@total_cost` in an instance variable, it might be wise to rename the `calc_total_cost` method `total_cost`. That way if the price, discount, or number of nights were ever to change, the total price would "update" automatically (update is in quotes because it was never stored in the first place).